CMSgov/mint-app
# add email to plan collaborator and add email domain to DB

## Changes and Description

- Add Email address to Plan Collaborator
- Add Email domain to the db to enforce valid email address/



## How to test this change
1. scripts/dev up:backend:debug && scripts/dev db:seed 
2. Try to make a collaborator with a bad email address
```gql
mutation createPlanCollaborator {
  createPlanCollaborator(input: {
      
    modelPlanID: "6e224030-09d5-46f7-ad04-4bb851b36eab"
    euaUserID: "BC0V"
    fullName:"Clay B"

    teamRole: LEADERSHIP
    email: "ahsdklashdlaksd"
    }) {
    id
    modelPlanID
    euaUserID
    fullName
    email
    teamRole
  }
}

```
3. Ensure the above failed
4.  Make a collaborator with a valid email address
```gql
mutation createPlanCollaborator {
  createPlanCollaborator(input: {
      
    modelPlanID: "6e224030-09d5-46f7-ad04-4bb851b36eab"
    euaUserID: "BC0V"
    fullName:"Clay B"

    teamRole: LEADERSHIP
    email: "clay@oddball.com"
    }) {
    id
    modelPlanID
    euaUserID
    fullName
    email
    teamRole
  }
}

```


## NOTE

This change likely breaks the front end. The add collaborator logic will need to be updated to pass email as required to create collaborator.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
